test/session_tests/DocumentStoreTest.ts

```diff
@@ -182,6 +182,16 @@ describe('Document store test', () => {
       expect(product['@metadata']['Raven-Node-Type']).to.not.exist;  
       expect(product['@metadata']['@collection']).to.equals('@empty');      
     });
+
+    it('shouldn\'t set id on plain object without id attribute defined', async () => {
+      let product: Product = <Product>{name: "New Product"};    
+
+      session = store.openSession({requestExecutor});      
+      await session.store<Product>(product);
+      await session.saveChanges();
+
+      expect(product.id).to.be.empty;      
```

Comment:
In C# if class does not have an id property defined we don't have anything to populate. We thought it would be good to have similar behavior in JS.